qa/restart-upgrade/src/test/java/org/opensearch/searchrelevance/bwc/restart/IndexMappingRestartIT.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.searchrelevance.bwc.restart;
+
+import static org.opensearch.searchrelevance.bwc.IndexMappingTestHelper.JUDGMENT_CACHE_INDEX;
+import static org.opensearch.searchrelevance.bwc.IndexMappingTestHelper.NEW_MAPPING;
+import static org.opensearch.searchrelevance.bwc.IndexMappingTestHelper.OLD_MAPPING;
+import static org.opensearch.searchrelevance.bwc.IndexMappingTestHelper.TEST_DOCUMENT;
+import static org.opensearch.searchrelevance.bwc.IndexMappingTestHelper.TEST_DOC_ID;
+
+import java.util.Map;
+
+import org.opensearch.searchrelevance.bwc.IndexMappingTestHelper;
+
+/**
+ * BWC (Backward Compatibility) Integration Test for Index Mapping Updates during full cluster restart.
+ *
+ * This test validates that index mappings are automatically updated when the entire cluster
+ * is restarted with a new plugin version that has bumped schema_version in the mapping JSON.
+ *
+ * Test scenario:
+ * 1. OLD cluster: Manually create judgment_cache index with old schema (version 0, no modelId/encodedPromptTemplate)
+ * 2. UPGRADED cluster (after full restart): Verify mapping has new fields and schema_version=1
+ */
+public class IndexMappingRestartIT extends AbstractSearchRelevanceRestartUpgradeTestCase {
+
+    /**
+     * Main BWC test for Index Mapping Updates during full cluster restart upgrade.
+     * Tests that:
+     * - OLD: Index is created with old mapping (schema_version 0, no modelId/encodedPromptTemplate)
+     * - UPGRADED: After full cluster restart, mapping is updated with new fields and schema_version=1
+     */
+    public void testIndexMappingUpdate_RestartUpgrade() throws Exception {
+        switch (getClusterType()) {
+            case OLD:
+                testCreateIndexWithOldMappingInOldCluster();
+                break;
+            case UPGRADED:
+                testValidateMappingUpdateInUpgradedCluster();
+                IndexMappingTestHelper.deleteIndex(client(), JUDGMENT_CACHE_INDEX, logger);
+                break;
+            default:
+                throw new IllegalStateException("Unknown cluster type: " + getClusterType());
+        }
+    }
+
+    /**
+     * OLD cluster: Create judgment_cache index with old schema (version 0).
+     * This simulates an index created before the schema_version was bumped.
+     */
+    private void testCreateIndexWithOldMappingInOldCluster() throws Exception {
+        // Create the judgment_cache index with old mapping (version 0)
+        IndexMappingTestHelper.createIndexWithMapping(client(), JUDGMENT_CACHE_INDEX, OLD_MAPPING, logger);
+
+        // Wait for cluster to propagate the index creation
+        Thread.sleep(2000);
```

Comment:
If in the api we are directly checking for status then we need this thread sleep here? If api status is 200 it means index is created.